yarn-project/circuits.js/src/structs/gas_used.ts

```diff
@@ -0,0 +1,36 @@
+import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
+import { type FieldsOf } from '@aztec/foundation/types';
+
+import { inspect } from 'util';
+
+import { type UInt32 } from './shared.js';
+
+/** Gas in each dimension used so far in the context of the current transaction or phase. */
+export class GasUsed {
```

Comment:
Just a heads up, when we actually publish these values we'll need to do a similar thing as `RevertCode` where we use different buffer lengths for serialization versus hashing.